### ISSUE TITLE
[Question] How can I have the config apply in a child directory, when commiting from a parent?

### ISSUE THREAD

**Issue by symonk** [pre-commit/pre-commit#1416]
we have something like this:

```
core_product:
    QA:
        auto_framework:
                      pre-commit-config.yaml
```

How, when commiting inside core_product can we have it use the `.pre-commit-config.yaml` from /QA/auto_framework/.pre-commit-config.yaml` ?

I tried doing pre-commit install --config `/QA/automation_framework/.pre-commit-config.yaml` but when commiting higher up, it states no `.pre-commit-config.yaml` found.

Is this possible? I set it all up using a virtualenv / makescript.  the install doesnt seem to honour the --config flag set, is there another way? (I dont want people to have to manually configure stuff after its been installed by me just to make it work)

**Comment by asottile** [comment]
can you show your exact `pre-commit install` command and your exact commit command (and output)?  it's unclear whether what you've posted is a typo or not (your folder structure for instance is missing a `.` and your install command has a leading slash)

it should work as you've described if the commands are correct -- that said, pre-commit works best when the configuration is at the default and expected location

**Comment by symonk** [comment]
My apologies, typo in the .pre-commit-config.yaml for sure, I will update when I’m not on mobile with something more reproducible.  Thanks

**Comment by symonk** [comment]
I guess my question in simpler terms is, does this work?

`pre-commit install --config path/to/config` ?  As far as I can tell it does not.  Here is an example tree:

```
Folder PATH listing
Volume serial number is 0E93-A7A4
C:.
├───.idea
│   │   .gitignore
│   │   markdown-navigator-enh.xml
│   │   markdown-navigator.xml
│   │   misc.xml
│   │   modules.xml
│   │   pre_commit.iml
│   │   workspace.xml
│   │
│   └───inspectionProfiles
│           profiles_settings.xml
│
└───tier_one
    └───tier_two
            .pre-commit-config.yaml
```

```
(pre_commit) C:\Users\sy\PycharmProjects\pre_commit>cd tier_one

(pre_commit) C:\Users\sy\PycharmProjects\pre_commit\tier_one>pre-commit install --config tier_two\.pre-commit-config.yaml
pre-commit installed at .git\hooks\pre-commit

(pre_commit) C:\Users\sy\PycharmProjects\pre_commit\tier_one>pre-commit run
An error has occurred: InvalidConfigError:
=====> .pre-commit-config.yaml does not exist
Check the log at C:\Users\sy/.cache\pre-commit\pre-commit.log
```

Is there anyway to be committing above tier_two here, e.g tier_one or the root dir, and have it automatically use the --config flag?

Thanks for your time @asottile and a great project here

```

**Comment by symonk** [comment]
The reason behind _why_ i need the above is, we have a structure to this where developers contribute to automated tests:

core_product
 -> automation_framework
    -> tests

devs typically commit in and around core_product, but occassionally are contributing to /tests, so I need a kind of 'global' solution to the problem as far as applying pre-commit goes

**Comment by asottile** [comment]
can you show a `git commit` it should just work -- your `pre-commit run` command needs `--config` as well if you're using the commandline tooling

**Comment by symonk** [comment]
@asottile sorry are you right sir, it seems OK! I have no idea why it doesn't in work, maybe because i'm installing from a makefile etc, i will have a look at that tomorrow.  Appreciate your time thanks.

is there anyway to ask pre-commit what its current config path is?

**Comment by asottile** [comment]
`pre-commit` itself doesn't store any state about the configuration and always takes it as a parameter, but the hook scripts do store that state:

```console
$ grep '^CONFIG =' .git/hooks/pre-commit
CONFIG = '.pre-commit-config.yaml'
```

**Comment by symonk** [comment]
When i do the install via my Makefile with the following:

```
bash -c "cd framework && python3.7 -m venv venv && source venv/bin/activate && pre-commit install --config=/Users/skerr/workspace/product/QA/framework/.pre-commit-config.yaml"
Running in migration mode with existing hooks at .git/hooks/pre-commit.legacy
Use -f to use only pre-commit.
pre-commit installed at .git/hooks/pre-commit
```

I had a look in the .git/hooks/pre-commit and there is no recollection of CONFIG = , just an ARGS which doesn't honour the path shown above:

```
 ARGS = ['hook-impl', '--config=QA/framework/.pre-commit-config.yaml', '--hook-type=pre-commit']
```

**Comment by symonk** [comment]
doesnt seem to honor this when installed through standard means:

```
(venv) ➜  essentials git:(epic/add-pre-commit-linting) ✗ pre-commit install --config ~/workspace/product/QA/framework/.pre-commit-config.yaml 
Running in migration mode with existing hooks at .git/hooks/pre-commit.legacy
Use -f to use only pre-commit.
pre-commit installed at .git/hooks/pre-commit
(venv) ➜  essentials git:(epic/add-pre-commit-linting) cat .git/hooks/pre-commit | grep ARGS
ARGS = ['hook-impl', '--config=QA/framework/.pre-commit-config.yaml', '--hook-type=pre-commit']
```

**Comment by asottile** [comment]
it resolves `--config` relative to the root of the repository since all git hooks execute from there -- I was confusing myself above a bit, it looks like the `CONFIG =` implementation detail is gone and now just `ARGS =`
